src/raft.c

```diff
@@ -341,23 +361,43 @@ void RaftExecuteCommandArray(RedisRaftCtx *rr,
         int old_entered_eval = rr->entered_eval;
 
         if ((cmdlen == 4 && !strncasecmp(cmd, "eval", 4)) ||
-            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7))) {
+            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7)) ||
+            (cmdlen == 5 && !strncasecmp(cmd, "fcall", 5))) {
             rr->entered_eval = 1;
         }
 
+        char *resp_call_fmt;
+
+        if (reply_ctx) {
+            if (cmds->acl) {
+                /* We have client so check ACL (C), return response in client format (0) and rm_call error as reply (E) */
+                resp_call_fmt = "CE0v";
+            } else {
+                resp_call_fmt = "0v";
+            }
+        } else {
+            if (cmds->acl) {
+                /* We don't have a client so only check ACL (C) */
+                resp_call_fmt = "Cv";
+            } else {
+                resp_call_fmt = "v";
+            }
+        }
```

Comment:
1- As we don't support older Redis versions, I think we can delete this: https://github.com/RedisLabs/redisraft/blob/722e48ec70fddde8fe3584d5e8ab6862dcdf8e6d/src/redisraft.c#L1931

2- Why don't we go with `CE0v` here all the time?

Reply:
calling `RM_call` with `C` flag when we have no ctx user will try to use client user, which is not what we want here.

Reply:
Hmm, can we at least do:

```c
char *resp_call_fmt = cmds->acl ? "CE0v" : "E0v";
```

Reply:
pretty sure I did get rid of it, don't see it here anymore.  confused.

Reply:
ok, tried to simplify it a bit with an explanation, also caught an issue I think